### DISCUSSION TITLE
How to install CMSIS_5 on STMCubeIDE 1.10.1?

### DISCUSSION THREAD

**Discussion by BitLoose** [ARM-software/CMSIS_5#1583] [General]
A number of developers want to install CMSIS on STM32CubeIDE as it isn't installing the DSP library for new projects anymore.
There are a few videos on installing CMSIS but all are well out of date, and determining the correct config for STMCubeIDE(Eclipse) could take some time and the usual painful head banging.
If anyone can solve our grief please post a reply!

**Comment by jkrech** [comment]
@BitLoose, please excuse my ignorance, but I am not familiar enough with STM32CubeIDE and how it installs/interacts with the "DSP library". Also the statement "it isn't installing the DSP library" isn't clear to me.
Maybe you could provide links to the documentation and highlight what has changed. Has STM32CubeIDE changed recently or is there changes to the CMSIS_5 repository that break the interoperability? E.g. the fact that the CMSIS-DSP Library has been moved to a separate repository?

**Reply by BitLoose** [reply]
Hi, well STMCubeIDE automatically installs tool chains and libraries when you start a new project, and although the latest version installs CMSIS it leaves out the DSP code, and there is no option to install it. Someone has now kindly sent me a link to an installation guide which I will try.
I've been spoilt using MS Visual Studio for development and debugging, and had hoped that these sort of problems would have been long since solved.  Eclipse IDE is even more primitive, especially in debugging, and the Arduino IDE is like a kids toy, sorry to say, but these apps should be far more developed by now. Moan over! :D  Thanks for taking the time to reply

**Reply by JonatanAntoni** [reply]
Hi @BitLoose,

I am not aware about how CubeIDE "installs" libraries. The official way is using Packs. The latest release of CMSIS is 5.9.0 and still contains the DSP code. The code was removed after that release. So if Cube gets the code directly from the GitHub development branch this would explain why its gone. If doing so, Cube would need to fetch DSP code explicitly from its new location.

Cheers,
Jonatan

**Reply by BitLoose** [reply]
STM should really have this in order by now, seems like a mess of incompatibilities, scripts and GUI´s tacked on, plus a good handful of bugs!
Cheers
